templates/batch-llm/main.py

```diff
@@ -75,15 +97,25 @@ def __call__(self, batch: Dict[str, np.ndarray]) -> Dict[str, list]:
 
 # Apply batch inference for all input data.
 ds = ray.data.read_text(INPUT_PATH)
+ds = ds.map(
+    construct_input_prompt,
+    fn_kwargs={"text_column": INPUT_TEXT_COLUMN},
+)
 ds = ds.map_batches(
     LLMPredictor,
     # Set the concurrency to the number of LLM instances.
     concurrency=num_llm_instances,
     # Specify the number of GPUs required per LLM instance.
     num_gpus=num_gpus_per_instance,
-    # Specify the batch size for inference. Set the batch size to as large possible without running out of memory.
-    # If you encounter CUDA out-of-memory errors, decreasing batch_size may help.
-    batch_size=10,
+    # Specify the batch size for inference. Set the batch size to as large
+    # as possible without running out of memory.
+    # If you encounter CUDA out-of-memory errors, decreasing
+    # batch_size may help.
+    batch_size=5,
```

Comment:
Any reason to reduce batch size?

Reply:
this was the switch to prevent stragglers where we would have some actors end up with more batches than others, setting the batch size to number of batches evenly divisible by number of actors (100 rows, 5 rows per batch, 4 actors -> each actor gets 5 batches of 5 rows)

Reply:
Got it. After fixing the `would have some actors end up with more batches than others` as we planned in https://docs.google.com/document/d/1vw5XIPqHK90O-7098kY4F3aDnuFxKfrpTPC8w2WG9mg/edit?usp=drive_open&ouid=104454598218723057493 , then we don't have this limitation, right?

Reply:
i think this is a slightly different issue than needing to call `repartition()` before `map_batches()`. In that case, the issue was that there as only 1 input block produced from reading 1 file, so we need to call `repartition(4)` to create enough blocks. This batch size reduction has a different effect, where we want to prevent some actors waiting around for others to finish. 

To illustrate with an example, with 100 input rows and 4 actors, and a batch size of 10:

Actor 1: (0, 10), (40, 50), (80, 90)
Actor 2: (10, 20), (50, 60), (90, 100)
Actor 3: (20, 30), (60, 70)
Actor 4: (30, 40), (70, 80)

In the above example, Actor 3 and 4 will need to wait for actor 1 and 2 to finish their third batches, since there aren't enough batches to go around. If we adjust the batch size to 5, we will have 20 total batches, with each actor getting 5 even batches.